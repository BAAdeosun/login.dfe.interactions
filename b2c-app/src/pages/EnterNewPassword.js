import React from 'react';
import components from '../components';
import { onChange, onError } from '../helpers/pageUpdatesHandler';

class EnterNewPassword extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            password: null,
            showErrors: false,
            errors: []
        }
        this.handleSubmit = this.handleSubmit.bind(this);
        this.onError = onError.bind(this);
        this.onChange = onChange.bind(this);
    }

    componentDidMount() {
        document.getElementById('api').style.display = 'none';
        document.title = 'Reset your password | National Careers Service';
    }

    handleSubmit(e) {
        e.preventDefault();
        //update error messages
        this.state.errors.forEach((error) => {
            error.visible.text = error.current.text;
            error.visible.showSummaryText = error.current.showSummaryText;
        });
        //do something to validate and decide if we submit or show errors
        if (this.state.password) {
            this.setState({ showErrors: false });
            //everything is valid, set data and submit B2C form
            this.setDataAndSubmit();
        }
        else {
            //show errors in each component
            this.setState({ showErrors: true });
        }
    }

    setDataAndSubmit() {
        //retrieve all elements we will need and set their values
        let b2cPassword = document.getElementById('newPassword');
        let b2cReenteredPassword = document.getElementById('reenterPassword');
        let b2cSubmitButton = document.getElementById('continue');

        if (b2cPassword && b2cReenteredPassword &&
            b2cSubmitButton) {
            b2cPassword.value = this.state.password;
            b2cReenteredPassword.value = this.state.password;
            //submit B2C form
            b2cSubmitButton.click();
        }
    }

    render() {
        return (
            <div id="enterNewPassword">

                <div className="govuk-width-container">
                    <components.Breadcrumbs />

                    <components.PageLevelErrorContainer errorItems={this.state.errors} summaryTextContent={<components.PasswordHelp />} />

                    <main className="govuk-main-wrapper">
                        <div className="govuk-grid-row">
                            <div className="govuk-grid-column-two-thirds">
                                <components.PageTitle size='xl' title="Reset your password" />

                                <form id="resetPasswordForm" onSubmit={this.handleSubmit} noValidate>
                                    <components.CreateNewPassword onChange={this.onChange} onError={this.onError} showErrors={this.state.showErrors} />
                                    <button className="govuk-button" id="preSubmit" type="submit">Reset password</button>
                                </form>

                            </div>
                        </div>
                    </main>

                </div>

            </div>
        )
    }
}

export default EnterNewPassword;