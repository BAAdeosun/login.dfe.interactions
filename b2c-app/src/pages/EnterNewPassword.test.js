import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import EnterNewPassword from './EnterNewPassword';

it('renders without crashing', () => {
    shallow(<EnterNewPassword />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<EnterNewPassword />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});