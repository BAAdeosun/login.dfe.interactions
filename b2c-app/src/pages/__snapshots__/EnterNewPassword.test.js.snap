// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="enterNewPassword"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <div
        className="pageLevelErrorContainer hide"
      >
        <div
          aria-labelledby="error-summary-title"
          className="govuk-error-summary"
          data-module="govuk-error-summary"
          role="alert"
          tabIndex="-1"
        >
          <h2
            className="govuk-error-summary__title"
            id="error-summary-title"
          >
            There is a problem
          </h2>
          <div
            className="govuk-error-summary__body"
          >
            <ul
              className="govuk-list govuk-error-summary__list"
              id="errorSummaryItems"
            />
          </div>
        </div>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              Reset your password
            </h1>
            <form
              id="customForm"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div>
                <div
                  className="govuk-form-group "
                >
                  <label
                    className="govuk-label"
                    htmlFor="newPasswordCustom"
                  >
                    Create new password
                  </label>
                  <input
                    className="govuk-input govuk-!-width-one-half"
                    id="newPasswordCustom"
                    name="newPassword"
                    noValidate={true}
                    onChange={[Function]}
                    type="password"
                  />
                </div>
                <div
                  className="govuk-form-group"
                >
                  <details
                    className="govuk-details"
                  >
                    <summary
                      className="govuk-details__summary"
                    >
                      <span
                        className="govuk-details__summary-text"
                      >
                        Help choosing a valid password
                      </span>
                    </summary>
                    <div
                      className="govuk-details__text"
                    >
                      <div>
                        <p
                          className="govuk-body"
                        >
                          Your password must:
                        </p>
                        <ul
                          className="govuk-list govuk-list--bullet"
                        >
                          <li>
                            be between 8 and 16 characters
                          </li>
                          <li>
                            include at least one uppercase letter and one lowercase letter
                          </li>
                          <li>
                            include at least one number or one symbol
                          </li>
                        </ul>
                      </div>
                    </div>
                  </details>
                </div>
                <div
                  className="govuk-form-group "
                >
                  <label
                    className="govuk-label"
                    htmlFor="reenteredPasswordCustom"
                  >
                    Re-type password
                  </label>
                  <input
                    className="govuk-input govuk-!-width-one-half"
                    id="reenteredPasswordCustom"
                    name="reenteredPassword"
                    noValidate={true}
                    onChange={[Function]}
                    type="password"
                  />
                </div>
              </div>
              <button
                className="govuk-button"
                id="preSubmit"
                type="submit"
              >
                Reset password
              </button>
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
